pkg/tempo/config.go

```diff
@@ -400,7 +398,19 @@ func formatPolicies(cfg []map[string]interface{}) ([]map[string]interface{}, err
 	return policies, nil
 }
 
-func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
+// Get satisfies the service.ParserProvider interface
+func (c *InstanceConfig) Get() (*config.Parser, error) {
```

Comment:
Possible improvement: we can verify this at compile-time by adding one of these checks: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/tailsamplingprocessor/sampling/always_sample.go#L26
With this the code will not compile if the struct doesn't satisfy the interface anymore.

It's a nice extra check but don't know if this is super common in Go 😅

Reply:
Nice suggestion, added in 383c8c1857e93f39ef7d379d063cbdb4bbdf2995